### PR TITLE
Bower to yarn updates when yarn.lock is present

### PR DESCRIPTION
**Pull request by biyeun**
https://dimagi-dev.atlassian.net/browse/SAAS-10881

##### ENVIRONMENTS AFFECTED
right now only staging should be affected, as it's the only branch with `yarn.lock`

### CODE REVIEW

File: src/commcare_cloud/fab/operations/staticfiles.py [hunk 1]
```diff
@@ -45,12 +60,28 @@ def bower_install():
...
 
+@parallel
+@roles(ROLES_STATIC + ROLES_DJANGO + ROLES_CELERY)
+def yarn_install():
+    yarn_lock = os.path.join(env.code_root, YARN_LOCK)
+    if not files.exists(yarn_lock):
+        return
+
+    with cd(env.code_root):
+        sudo('npm install -g yarn')
```
Comment (biyeun) [inline src/commcare_cloud/fab/operations/staticfiles.py:81]: @dannyroberts this seems to fail with 
```
out: npm WARN checkPermissions Missing write access to /usr/local/lib/node_modules
out: npm ERR! path /usr/local/lib/node_modules
out: npm ERR! code EACCES
out: npm ERR! errno -13
out: npm ERR! syscall access
out: npm ERR! Error: EACCES: permission denied, access '/usr/local/lib/node_modules'
out: npm ERR!  { [Error: EACCES: permission denied, access '/usr/local/lib/node_modules']
out: npm ERR!   stack:
out: npm ERR!    'Error: EACCES: permission denied, access \'/usr/local/lib/node_modules\'',
out: npm ERR!   errno: -13,
out: npm ERR!   code: 'EACCES',
out: npm ERR!   syscall: 'access',
out: npm ERR!   path: '/usr/local/lib/node_modules' }
```

any ideas?

File: src/commcare_cloud/fab/operations/offline.py [hunk 2]
```diff
@@ -35,7 +36,13 @@ def prepare_files():
     })))
 
     # NPM
-    local('cd {}/commcare-hq && npm install --production'.format(OFFLINE_STAGING_DIR))
+    local('cd {}/commcare-hq'.format(OFFLINE_STAGING_DIR))
+
+    yarn_lock = os.path.join(env.code_root, YARN_LOCK)
+    if files.exists(yarn_lock):
```
Comment (orangejenny) [inline src/commcare_cloud/fab/operations/offline.py:42]: I might have missed this in another PR description, but is the plan to replace both bower and npm with yarn?

Comment (biyeun) [reply src/commcare_cloud/fab/operations/offline.py:42]: yes, `yarn` runs on top of `npm` and effectively works with both `npm` and `bower` repositories

Comment (orangejenny) [reply src/commcare_cloud/fab/operations/offline.py:42]: So we're modernizing *and* using one fewer tool? Fantastic.

Comment (biyeun) [reply src/commcare_cloud/fab/operations/offline.py:42]: yes, except `yarn 2` and `npm 2` are on the way...but baby steps...

**Pull request merged**